### PR TITLE
add support for node specifiers

### PR DESCRIPTION
**Pull request by khanayan123**
### What does this PR do?
adds support for fastify versions >= 4.23.0 by adding support for node:* specifiers. Also updates addHook and register functionality to make sure that double patching doesn't occur when instrumenting node:* specifiers. For example instrumenting both dns and node:dns causes double patching

### Motivation
fastify instrumentation breaking on versions >= 4.23.0

### Notes
this PR is a continuation of this https://github.com/DataDog/dd-trace-js/pull/3676

### CODE REVIEW

File: packages/datadog-instrumentations/src/helpers/instrument.js [hunk 1]
```diff
@@ -21,11 +21,17 @@ exports.channel = function (name) {
...
   }
 
-  instrumentations[name].push({ name, versions, file, hook })
+  name.forEach(val => {
+    if (!instrumentations[val]) {
+      instrumentations[val] = []
+    }
+
+    instrumentations[val].push({ name: val, versions, file, hook })
+  })
```
Comment (Qard) [inline packages/datadog-instrumentations/src/helpers/instrument.js:34]: I would generally recommend for-of over forEach.

```suggestion
  for (const val of name) {
    if (!instrumentations[val]) {
      instrumentations[val] = []
    }

    instrumentations[val].push({ name: val, versions, file, hook })
  }
```

File: packages/datadog-instrumentations/src/net.js [hunk 2]
```diff
@@ -17,8 +17,14 @@ const errorTCPCh = channel('apm:net:tcp:error')
...
-addHook({ name: 'net' }, net => {
-  require('dns')
+const names = ['net', 'node:net']
+
+addHook({ name: names }, (net, version, name) => {
+  if (name === 'net') {
+    require('dns')
+  } else {
+    require('node:dns')
+  }
```
Comment (Qard) [inline packages/datadog-instrumentations/src/net.js:27]: Why are we loading the dns module here again? Maybe we could use a comment to remember?

**Review by Qard — DISMISSED** [review]
Some minor nits but otherwise LGTM.

File: packages/datadog-instrumentations/src/net.js (hunk 2 above)
Comment (khanayan123) [reply packages/datadog-instrumentations/src/net.js:29]: I believe it's because otherwise net gets an uninstrumented instance & thus we miss the dns calls

**Review by rochdev — APPROVED** [review]

**Pull request merged**